Add machine-readable property definitions to repository (#445)
Materials-Consortia/OPTIMADE: optimade.rst

          },
          "lattice_vectors": {
            "$id": "urn:uuid:81edf372-7b1b-4518-9c14-7d482bd67834",
            "title": "Lattice vectors",
            "x-optimade-definition": {

>               "label": "lattice_vectors_optimade_structures"

Reviewer: ```suggestion
              "label": "lattice_vectors_optimade_structures",
```

              "kind": "property",
              "format": "1.2",
              "version": "1.2.0",
              "name": "lattice_vectors"
            },
            "x-optimade-type": "list",
            "x-optimade-dimensions": {

>                "names": ["dim_lattice", "dim_spatial"]

Reviewer: ```suggestion
               "names": ["dim_lattice", "dim_spatial"],
```

A Property Definition MUST be composed according to the combination of the requirements in the subsection `Property Definition keys from JSON Schema`_ below and the following additional requirements:

**REQUIRED keys for the outermost level of the Property Definition and OPTIONAL for other levels:**

- :field:`$id`: String, :field:`$schema`: String, :field:`title`: String, and :field:`description`: String.
  See the subsection `Property definition keys from JSON Schema`_ for the definitions of these fields.
  They are defined in that subsection as OPTIONAL on any level of the Property Definition, but are REQUIRED on the outermost level.

.. _definition of the x-optimade-definition field:

- :field:`x-optimade-definition`: Dictionary.
  Additional information about the definition that is not covered by fields in the JSON Schema standard.

  **REQUIRED keys:**

>   .. _definition of the property-format field:

Reviewer 2: ```suggestion
.. _definition of the property-format field:
```

To make a property definition expressible in any output format, the fields of the property definition below are specified using OPTIMADE data types.
When a property definition is communicated using a specific data format (e.g., JSON), the property definition is implemented in that data format by mapping the OPTIMADE data types into the corresponding data types for that output format.

> Clients are meant to be able to rely on that properties with the same :field:`$id` fields represents equivalently defined properties.
> Hence, when a Property Definition that has been published previously is updated, it is of major importance to decide if the update merely amend, annotate, or clarify the definition in a way that leaves it functionally the same and thus can retain the :field:`$id`, or whether the property is redefined.

Reviewer 3: ```suggestion
Clients are meant to be able to rely on the fact that properties with the same :field:`$id` fields represents equivalently defined properties.
Hence, when a Property Definition that has been published previously is updated, it is of major importance to decide if the updates merely amend, annotate, or clarify the definition in a way that leaves it functionally the same and thus can retain the :field:`$id`, or whether the property is redefined.
```

    Values can only be defined to be a single OPTIMADE data type or, optionally, :val:`null`.
    This restriction is intended to reduce the complexity of possible data types that implementations have to handle in different formats and database backends.

**Keys that are REQUIRED on the outermost level of a Property Definition, but otherwise OPTIONAL:**

- :field:`$schema`: String.
  A URL for a meta schema that describes the Property Definitions format.

>   For Property Definitions adhering to the format described in this document, it should be set to: "https://schemas.optimade.org/meta/v1.2/optimade/property_definition.json".

Reviewer: ```suggestion
  For Property Definitions adhering to the format described in this document, it should be set to: :val:`https://schemas.optimade.org/meta/v1.2/optimade/property_definition.json`.
```

  **OPTIONAL**

  - :field:`uniqueItems`: Boolean.
    If :val:`TRUE`, the defined property is an array that MUST only contain unique items.
    If :val:`FALSE`, this field sets no limitation on the defined property.

  Furthermore, despite being defined in the JSON Schema standard, the fields :field:`minItems` and :field:`maxItems` MUST NOT be used to indicate limits of the number of items of a list.

>   See the `definition of the x-optimade-dimension field`_ for more information.

Reviewer 2: ```suggestion
  See the `definition of the x-optimade-dimensions field`_ for more information.
```

Each symbol can also be suffixed by a single ``^`` character followed by a positive or negative integer to indicate the power of the preceding symbol, e.g., ``m^3`` for cubic meter, ``m^-3`` for inverse cubic meter.
(Positive integers MUST NOT be preceded by a plus sign.)
Each unit or constant symbol MAY be directly prefixed by a prefix symbol.
A prefix symbol MUST be directly followed by a unit symbol, i.e., it MUST NOT be used on its own, and MUST NOT be followed by ``^`` to indicate a power.
When defining prefix symbols it is important to ensure that they do not introduce ambiguity.
If there are ambiguous interpretations of a symbol as either having or not having a prefix, it MUST be interpreted as a unit without a prefix.

Furthermore:

- No whitespace, parentheses, or other symbols than specified above are permitted.
- The (prefixed) unit and constant symbols MUST appear in alphabetical order.

Physical Unit Definitions
-------------------------

An OPTIMADE Physical Unit Definition is a dictionary adhering to the following format:

**REQUIRED keys:**

- :field:`$schema`: String.
  A URL for a meta schema that describes the Physical Unit Definitions format.

>   For Property Definitions adhering to the format described in this document, it should be set to: "https://schemas.optimade.org/meta/v1.2/optimade/physical_unit_definition.json".

Reviewer: ```suggestion
  For Property Definitions adhering to the format described in this document, it should be set to: :val:`https://schemas.optimade.org/meta/v1.2/optimade/physical_unit_definition.json`.
```
